.gitignore

```diff
@@ -13,3 +13,5 @@ coverage.html
 .glide
 .scBuildImage
 .init
+.generate*
+**/zz*.go
```

Comment:
some history and current state on generated code, from the peanut gallery.

Historically (in the main kube repo) we *had* to commit  generated code, because `go get` was important, because the only real API client was in the same repo.  Nowadays, the client snapshots all it needs into a new repo, so `go get` can take a long walk off a short pier (for the main repo).

Do you need this repo to be `go get` compatible?  If so, you pretty much have to check in the generated code.

The alternative to committing them is to have fairly complicated builds that auto-detect when to rebuild them.  The main repo has a (perhaps overly complicated) solution for this, wrt a handful of code-generators, but not all.  I see this PR has a less "sophisticated" build here, which seems OK at first blush.

So the main reason that zz_generated.* is checked into the main repo is that I have not resurrected the PR to remove them.  I will do once 1.5.x cherrypicks settle down.

So philosophically I *want* to un-commit generated code.  It's insane how much rebasing happens because of it.  It's not as simple (for the main repo).  If you need `go get` or if you start to do the JSON-generation or proto-generation stuff, you will get ahead of my work, and be forced to check it in (or solve the build yourself).  I'd be happy for help, but it's a massive distraction from "productive" work.

Looking at this PR, it's obvious that even the work we have done so far wrt deepcopy is not done enough.

TL;DR: no, go back and read what I wrote.  It's too complicated to summarize.